### DISCUSSION TITLE
Why is the "typeof(a).name" function appearing in the documentation but not searchable in vlib?

### DISCUSSION THREAD

**Discussion by tzSharing** [vlang/v#14022] [General]
```
fn main(){
    length := 1.0
    println(typeof(length).name)
}
----------------
f64
```
This function is very useful to know the data type of a variable, but I didn't find it in vlib.
Without reading the official documentation, I wouldn't have even known it existed.(maybe more  ^__^ )

**Comment by ghost** [comment]
docs.md is written on a volunteer basis and is not supervised by anyone in particular.
It is just that no one describes it.
You can add an explanation for it.

However, it is not clear to what extent this should be detailed in the documentation.
I don't think that is currently stipulated.

**Comment by ntrel** [comment]
`typeof` is not a runtime function, so it's not in `vlib`. `typeof` should produce a type, not a value, so it could be used in declarations:

```v
struct S1(T) {
    v typeof(T.some_field)
}
```
That will hopefully be implemented later.

`T.name` gives a string with the type name. A `typeof` expression produces a type so the `.name` is a type property.

This follows other statically-typed languages with strong support for generic programming.

**Reply by tzSharing** [reply]
Ok, but I hope there will be a dedicated page in the future to search all aspects of v-language knowledge points (including documentation, books, standard libraries, third-party libraries, API references...)

**Reply by JalonSolov** [reply]
There's are 2 books on V, one that was written a couple of years ago, and only available in Japanese, and the other is linked on the home page.

There will be more documentation as V matures.
